sql/expression/function/str_to_date_test.go

```diff
@@ -18,9 +18,17 @@ func TestStrToDate(t *testing.T) {
 		name     string
 		dateStr  string
 		fmtStr   string
-		expected string
+		expected interface{}
 	}{
 		{"standard", "Dec 26, 2000 2:13:15", "%b %e, %Y %T", "2000-12-26 02:13:15"},
+		{"ymd", "20240101", "%Y%m%d", "2024-01-01"},
+		{"ymd", "2024121", "%Y%m%d", "2024-12-01"},
+		{"ymd", "20241301", "%Y%m%d", nil},
+		{"ymd", "20240001", "%Y%m%d", nil},
+		{"ymd-with-time", "2024010203:04:05", "%Y%m%d%T", "2024-01-02 03:04:05"},
+		{"ymd-with-time", "202408122:03:04", "%Y%m%d%T", "2024-08-12 02:03:04"},
+		// TODO: It shoud be nil, but returns "2024-02-31"
```

Comment:
Are you going to fix this test case? I think it's ok to leave as a TODO.

Reply:
It's not a big problem. So I think it can be left as TODO.